diff --git a/src/TalonOne.Test/Api/ManagementApiTests.cs b/src/TalonOne.Test/Api/ManagementApiTests.cs
index 7aa10d5..253db7a 100644
--- a/src/TalonOne.Test/Api/ManagementApiTests.cs
+++ b/src/TalonOne.Test/Api/ManagementApiTests.cs
@@ -145,7 +145,7 @@ namespace TalonOne.Test.Api
             NewAttribute newAttribute = default;
             var response = await _instance.CreateAttributeAsync(newAttribute);
             var model = response.Created();
-            Assert.IsType<Attribute>(model);
+            Assert.IsType<TalonOne.Model.Attribute>(model);
         }
 
         /// <summary>
@@ -1065,7 +1065,7 @@ namespace TalonOne.Test.Api
             long attributeId = default;
             var response = await _instance.GetAttributeAsync(attributeId);
             var model = response.Ok();
-            Assert.IsType<Attribute>(model);
+            Assert.IsType<TalonOne.Model.Attribute>(model);
         }
 
         /// <summary>
@@ -2405,7 +2405,7 @@ namespace TalonOne.Test.Api
             NewAttribute newAttribute = default;
             var response = await _instance.UpdateAttributeAsync(attributeId, newAttribute);
             var model = response.Ok();
-            Assert.IsType<Attribute>(model);
+            Assert.IsType<TalonOne.Model.Attribute>(model);
         }
 
         /// <summary>
diff --git a/src/TalonOne/Client/ClientUtils.cs b/src/TalonOne/Client/ClientUtils.cs
index 3bc910b..ceb10da 100644
--- a/src/TalonOne/Client/ClientUtils.cs
+++ b/src/TalonOne/Client/ClientUtils.cs
@@ -42,14 +42,6 @@ namespace TalonOne.Client
         /// </summary>
         public enum ApiKeyHeader
         {
-            /// <summary>
-            /// The Authorization header
-            /// </summary>
-            Authorization,
-            /// <summary>
-            /// The Authorization header
-            /// </summary>
-            Authorization,
             /// <summary>
             /// The Authorization header
             /// </summary>
@@ -66,10 +58,6 @@ namespace TalonOne.Client
         {
             switch(value)
             {
-                case ApiKeyHeader.Authorization:
-                    return "Authorization";
-                case ApiKeyHeader.Authorization:
-                    return "Authorization";
                 case ApiKeyHeader.Authorization:
                     return "Authorization";
                 default:
@@ -207,12 +195,12 @@ namespace TalonOne.Client
                 return ApplicationSession.StateEnumToJsonValue(applicationSessionStateEnum);
             if (obj is AsyncCouponsData.NotificationTypeEnum asyncCouponsDataNotificationTypeEnum)
                 return AsyncCouponsData.NotificationTypeEnumToJsonValue(asyncCouponsDataNotificationTypeEnum);
-            if (obj is Attribute.EntityEnum attributeEntityEnum)
-                return Attribute.EntityEnumToJsonValue(attributeEntityEnum);
-            if (obj is Attribute.TypeEnum attributeTypeEnum)
-                return Attribute.TypeEnumToJsonValue(attributeTypeEnum);
-            if (obj is Attribute.AllowedSubscriptionsEnum attributeAllowedSubscriptionsEnum)
-                return Attribute.AllowedSubscriptionsEnumToJsonValue(attributeAllowedSubscriptionsEnum);
+            if (obj is TalonOne.Model.Attribute.EntityEnum attributeEntityEnum)
+                return TalonOne.Model.Attribute.EntityEnumToJsonValue(attributeEntityEnum);
+            if (obj is TalonOne.Model.Attribute.TypeEnum attributeTypeEnum)
+                return TalonOne.Model.Attribute.TypeEnumToJsonValue(attributeTypeEnum);
+            if (obj is TalonOne.Model.Attribute.AllowedSubscriptionsEnum attributeAllowedSubscriptionsEnum)
+                return TalonOne.Model.Attribute.AllowedSubscriptionsEnumToJsonValue(attributeAllowedSubscriptionsEnum);
             if (obj is BaseCampaign.FeaturesEnum baseCampaignFeaturesEnum)
                 return BaseCampaign.FeaturesEnumToJsonValue(baseCampaignFeaturesEnum);
             if (obj is BaseCampaign.StateEnum baseCampaignStateEnum)
diff --git a/src/TalonOne/Model/BestPriorPriceRequest.cs b/src/TalonOne/Model/BestPriorPriceRequest.cs
index 8448eb8..4733948 100644
--- a/src/TalonOne/Model/BestPriorPriceRequest.cs
+++ b/src/TalonOne/Model/BestPriorPriceRequest.cs
@@ -177,7 +177,7 @@ namespace TalonOne.Model
                             timeframeEndDate = new Option<DateTime?>(JsonSerializer.Deserialize<DateTime>(ref utf8JsonReader, jsonSerializerOptions));
                             break;
                         case "timeframe":
-                            timeframe = new Option<int?>(utf8JsonReader.GetString());
+                            timeframe = new Option<int?>(utf8JsonReader.TokenType == JsonTokenType.Null ? (int?)null : utf8JsonReader.GetInt32());
                             break;
                         case "strictEndDate":
                             strictEndDate = new Option<bool?>(utf8JsonReader.TokenType == JsonTokenType.Null ? (bool?)null : utf8JsonReader.GetBoolean());
@@ -255,7 +255,7 @@ namespace TalonOne.Model
             JsonSerializer.Serialize(writer, bestPriorPriceRequest.Skus, jsonSerializerOptions);
             writer.WriteString("timeframeEndDate", bestPriorPriceRequest.TimeframeEndDate.ToString(TimeframeEndDateFormat));
 
-            writer.WriteString("timeframe", bestPriorPriceRequest.Timeframe);
+            writer.WriteNumber("timeframe", (long)bestPriorPriceRequest.Timeframe);
 
             writer.WriteBoolean("strictEndDate", bestPriorPriceRequest.StrictEndDate);
 
