diff --git a/src/TalonOneSdk.Test/Api/ManagementApiTests.cs b/src/TalonOneSdk.Test/Api/ManagementApiTests.cs
index 7aa10d5..253db7a 100644
--- a/src/TalonOneSdk.Test/Api/ManagementApiTests.cs
+++ b/src/TalonOneSdk.Test/Api/ManagementApiTests.cs
@@ -145,7 +145,7 @@ namespace TalonOneSdk.Test.Api
             NewAttribute newAttribute = default;
             var response = await _instance.CreateAttributeAsync(newAttribute);
             var model = response.Created();
-            Assert.IsType<Attribute>(model);
+            Assert.IsType<TalonOneSdk.Model.Attribute>(model);
         }

         /// <summary>
@@ -1065,7 +1065,7 @@ namespace TalonOneSdk.Test.Api
             long attributeId = default;
             var response = await _instance.GetAttributeAsync(attributeId);
             var model = response.Ok();
-            Assert.IsType<Attribute>(model);
+            Assert.IsType<TalonOneSdk.Model.Attribute>(model);
         }

         /// <summary>
@@ -2405,7 +2405,7 @@ namespace TalonOneSdk.Test.Api
             NewAttribute newAttribute = default;
             var response = await _instance.UpdateAttributeAsync(attributeId, newAttribute);
             var model = response.Ok();
-            Assert.IsType<Attribute>(model);
+            Assert.IsType<TalonOneSdk.Model.Attribute>(model);
         }

         /// <summary>
diff --git a/src/TalonOneSdk/Client/ClientUtils.cs b/src/TalonOneSdk/Client/ClientUtils.cs
index 3bc910b..ceb10da 100644
--- a/src/TalonOneSdk/Client/ClientUtils.cs
+++ b/src/TalonOneSdk/Client/ClientUtils.cs
@@ -42,14 +42,6 @@ namespace TalonOneSdk.Client
         /// </summary>
         public enum ApiKeyHeader
         {
-            /// <summary>
-            /// The Authorization header
-            /// </summary>
-            Authorization,
-            /// <summary>
-            /// The Authorization header
-            /// </summary>
-            Authorization,
             /// <summary>
             /// The Authorization header
             /// </summary>
@@ -66,10 +58,6 @@ namespace TalonOneSdk.Client
         {
             switch(value)
             {
-                case ApiKeyHeader.Authorization:
-                    return "Authorization";
-                case ApiKeyHeader.Authorization:
-                    return "Authorization";
                 case ApiKeyHeader.Authorization:
                     return "Authorization";
                 default:
diff --git a/src/TalonOneSdk/Model/NewCustomerSessionV2.cs b/src/TalonOneSdk/Model/NewCustomerSessionV2.cs
index 95997ea..a5fe0f3 100644
--- a/src/TalonOneSdk/Model/NewCustomerSessionV2.cs
+++ b/src/TalonOneSdk/Model/NewCustomerSessionV2.cs
@@ -576,8 +576,11 @@ namespace TalonOneSdk.Model
                 writer.WritePropertyName("loyaltyCards");
                 JsonSerializer.Serialize(writer, newCustomerSessionV2.LoyaltyCards, jsonSerializerOptions);
             }
-            var stateRawValue = NewCustomerSessionV2.StateEnumToJsonValue(newCustomerSessionV2.StateOption.Value.Value);
-            writer.WriteString("state", stateRawValue);
+            if (newCustomerSessionV2.StateOption.IsSet)
+            {
+                var stateRawValue = NewCustomerSessionV2.StateEnumToJsonValue(newCustomerSessionV2.StateOption.Value.Value);
+                writer.WriteString("state", stateRawValue);
+            }
             if (newCustomerSessionV2.CartItemsOption.IsSet)
             {
                 writer.WritePropertyName("cartItems");
diff --git a/src/TalonOneSdk/Model/CampaignRulesetChangedNotificationItem.cs b/src/TalonOneSdk/Model/CampaignRulesetChangedNotificationItem.cs
index aad7aa6..2047417 100644
--- a/src/TalonOneSdk/Model/CampaignRulesetChangedNotificationItem.cs
+++ b/src/TalonOneSdk/Model/CampaignRulesetChangedNotificationItem.cs
@@ -166,10 +166,10 @@ namespace TalonOneSdk.Model
                             campaign = new Option<Object>(JsonSerializer.Deserialize<Object>(ref utf8JsonReader, jsonSerializerOptions));
                             break;
                         case "oldRuleset":
-                            oldRuleset = new Option<Ruleset>(utf8JsonReader.GetString());
+                            oldRuleset = new Option<Ruleset>(JsonSerializer.Deserialize<Ruleset>(ref utf8JsonReader, jsonSerializerOptions));
                             break;
                         case "ruleset":
-                            ruleset = new Option<Ruleset>(utf8JsonReader.GetString());
+                            ruleset = new Option<Ruleset>(JsonSerializer.Deserialize<Ruleset>(ref utf8JsonReader, jsonSerializerOptions));
                             break;
                         default:
                             break;
@@ -238,10 +238,16 @@ namespace TalonOneSdk.Model
             else
                 writer.WriteNull("campaign");
             if (campaignRulesetChangedNotificationItem.OldRulesetOption.IsSet)
-                writer.WriteString("oldRuleset", campaignRulesetChangedNotificationItem.OldRuleset);
+            {
+                writer.WritePropertyName("oldRuleset");
+                JsonSerializer.Serialize(writer, campaignRulesetChangedNotificationItem.OldRuleset, jsonSerializerOptions);
+            }

             if (campaignRulesetChangedNotificationItem.RulesetOption.IsSet)
-                writer.WriteString("ruleset", campaignRulesetChangedNotificationItem.Ruleset);
+            {
+                writer.WritePropertyName("ruleset");
+                JsonSerializer.Serialize(writer, campaignRulesetChangedNotificationItem.Ruleset, jsonSerializerOptions);
+            }
         }
     }
 }
